File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0
        for idx, num in enumerate(nums):
            if val == num:
                continue
            nums[k], nums[idx] = nums[idx], nums[k]
            k += 1
        return k

        

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // for num, check if its difference is in the hashmap, if yes, YAY, if no, add
        std::unordered_map<int, int> mymap;
        for (int idx{}; idx < nums.size(); ++idx){
            if (mymap.count(target - nums[idx])) return {mymap[target - nums[idx]], idx};
            mymap.insert({nums[idx], idx});
        }
        return {};
    }
};
